xwiki-authenticator-trusted-adapters/xwiki-authenticator-trusted-attributes/src/main/java/org/xwiki/contrib/authentication/AttributesTrustedAuthenticationAdapter.java

```diff
@@ -0,0 +1,193 @@
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.web.XWikiRequest;
+
+/**
+ * Implementation of the {@link TrustedAuthenticationAdapter} for attributes authentication.
+ *
+ * @since 1.9.0
+ */
+@Component
+@Singleton
+@Named("attributes")
+public class AttributesTrustedAuthenticationAdapter implements TrustedAuthenticationAdapter
+{
+    // Configuration
+    private static final String CONFIG_AUTH_FIELD = "auth_field";
+
+    private static final String CONFIG_ID_FIELD = "id_field";
+
+    private static final String CONFIG_GROUP_FIELD = "group_field";
+
+    private static final String CONFIG_GROUP_VALUE_SEPARATOR = "group_value_separator";
+
+    private static final String CONFIG_LOGOUT_URL = "logout_url";
+
+    private static final String CONFIG_ATTRIBUTE_ENCODING = "attribute_encoding";
+
+    // Default values for configuration
+    private static final String DEFAULT_AUTH_FIELD = "remote_user";
+
+    private static final String DEFAULT_ID_FIELD = DEFAULT_AUTH_FIELD;
+
+    private static final String DEFAULT_GROUP_VALUE_SEPARATOR = "\\|";
+
+    private static final String LOGOUT_URL_REDIRECTION_PLACEHOLDER = "__REDIRECT__";
+
+    private static final char COMMA_SEPARATOR = ',';
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private TrustedAuthenticationConfiguration configuration;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Override
+    public String getUserUid()
+    {
+        // There is no need for a Secret field when attributes.
+        // Secret field cross checking is a pseudo security fix for the relative less secure HTTP headers.
+        return getAttribute(configuration.getCustomProperty(CONFIG_AUTH_FIELD, DEFAULT_AUTH_FIELD));
+    }
+
+    @Override
+    public String getUserName()
+    {
+        return getAttribute(configuration.getCustomProperty(CONFIG_ID_FIELD, DEFAULT_ID_FIELD));
+    }
+
+    @Override
+    public String getUserProperty(String name)
+    {
+        return getAttribute(name);
+    }
+
+    @Override
+    public boolean isUserInRole(String role)
+    {
+        return getUserRoles().contains(role);
+    }
+
+    /**
+     * @param name the name of the attribute.
+     * @return the value of the named request attribute, or null if no value is defined.
+     */
+    private String getAttribute(String name)
+    {
+        if (StringUtils.isBlank(name)) {
+            return null;
+        }
+
+        XWikiRequest request = contextProvider.get().getRequest();
+        if (request == null) {
+            return null;
+        }
+
+        String value = (String) request.getAtrribute(name);
+
+        if (StringUtils.isNotBlank(value)) {
+            String encoding = configuration.getCustomProperty(CONFIG_ATTRIBUTE_ENCODING, null);
+            if (StringUtils.isNotBlank(encoding) && Charset.isSupported(encoding)) {
+                try {
+                    value = new String(value.getBytes("ISO-8859-1"), encoding);
```

Comment:
It's not very clear to me why this is needed, it's supposed to be parsed already. At the very least, it deserves a comment to explain it.

Reply:
Ah, am not sure if there needs to be a decoding akin to how Headers Adapter is done. What do you think? If it is not needed, please feel free to remove it

Reply:
Happy to remove this section if it makes no sense. What do you think?

Reply:
As I said, I don't understand why it would be needed, but there must be a reason why you did this in the first place ?

Reply:
Perfect, that does simplify things. I've added a new commit.

Also added more safety around getAttribute casting (as it returns an Object and not String)